### ISSUE TITLE
Extremely flaky resources form test 

### ISSUE THREAD

**Issue by cesarvarela** [responsible-ai-collaborative/aiid#823]
Though test flakiness has stabilized, for the most part, the test `Should show the taxonomy form of resources` is still flaky.

An ongoing issue on Cypress may be related: https://github.com/cypress-io/cypress/issues/7306 and implementing a workaround like the one suggested https://github.com/cypress-io/cypress/issues/7306#issuecomment-1076451070 may be worth a try.

The cause may be something that has anything to do with the issue above.

 
![image](https://user-images.githubusercontent.com/1172479/179824602-7e77852c-3160-4f06-8388-c84c9665d4a4.png)

**Comment by lmcnulty** [comment]
I was blocked for a while by this test failing for seemingly no reason, and I found that it was because in `cite.js`, when the `classificationsArray` is empty, the rendering of taxonomies is conditional on the login state:

```jsx
{taxonomies.map((t) => {
  const canEdit =
    isRole('taxonomy_editor') ||
    isRole('taxonomy_editor_' + t.namespace.toLowerCase());

  return canEdit || t.classificationsArray.length > 0 ? (
    <Taxonomy
      key={t.namespace}
      taxonomy={t}
      incidentId={incident.incident_id}
      canEdit={canEdit}
    />
  ) : null;
})}
```

The server-side rendering acts as if the user is not logged in, so a re-render has to be triggered for the resources taxonomy to appear. [I got it to pass](https://github.com/responsible-ai-collaborative/aiid/pull/761/commits/a1b2021ce932439a6cf451ba88f07e50ef35ce41) by adding a `useEffect` that updates the state on the initial render.

I'm not sure why this would result in flakiness rather than guaranteed failure, but it seems worth mentioning.
